bundles/specmate-emfrest/src/com/specmate/emfrest/authentication/Login.java

```diff
@@ -0,0 +1,60 @@
+package com.specmate.emfrest.authentication;
+
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.log.LogService;
+
+import com.specmate.auth.api.IAuthentificationService;
+import com.specmate.common.SpecmateException;
+import com.specmate.emfrest.api.IRestService;
+import com.specmate.emfrest.api.RestServiceBase;
+
+@Component(service = IRestService.class)
+public class Login extends RestServiceBase {
+	public static final String SERVICE_NAME = "login";
+	private static final String USERNAME_PARAM = "username";
+	private static final String PASSWORD_PARAM = "password";
+	private static final String PROJECT_PARAM = "projectname";
+
+	private IAuthentificationService authService;
+	private LogService logService;
+	
+	@Override
+	public String getServiceName() {
+		return SERVICE_NAME;
+	}
+	
+	@Override
+	public boolean canGet(Object target) {
+		return true;
+	}
+	
+	@Override
+	public Object get(Object object, MultivaluedMap<String, String> queryParams) {
+		String username = queryParams.getFirst(USERNAME_PARAM);
```

Comment:
I'm just thinking: Is it right to send username and password? For now, the traffic between client and server is not encrypted. We definitely hav to change this as soon as we deploy this.

Reply:
Using SSL, the query parameters are encrypted. Having said that, confidential data can still leak, e.g. in web-server logs. Hence, I agree that sending plain text passwords, even over SSL, is not a good idea.
I will improve the implementation and try to follow the design of [Digest access authentication](https://en.wikipedia.org/wiki/Digest_access_authentication).

Reply:
The problem with this will be that we need username and password (plain) to authenticate against the hp-tools.

Reply:
Hm, allright, then I understand. Ok, then we MUST use SSL. To diminish the chance of data leakage to log files, we send username/password in the http header instead of the URL. Not optimal, but best I can think of now.

Reply:
Yeah, this is not optimal, but the only solution for now, I guess. We could encrypt u/p with a key exchanged using Diffie-Hellman. Is that overkill?

Reply:
Sorry, what I meant was to make a POST request, instead of sending the credentials in the URL.

Reply:
Yes, asymmetric encryption would work. The client uses the public key to encrypt the data. The server decrypts it with the private key. We don't need key exchanges, AFAICS.

Reply:
Assymetric encryption would only work, if we do not store the private key in the repository. I was thinking about encrypting symetrically, but using a secret key exchanged using Diffie-Hellman. But I think the assymetric version is better, because it is just easier. We have to make the keys, however, configurable.

Reply:
I think we should choose a protocol as secure (but not less) as the other systems we communicate with. We can discuss this through other channels :)